doc/Programming-guide/Using-Kaa-endpoint-SDKs/C++/SDK-Edison/index.md

```diff
@@ -81,71 +84,108 @@ Refer to [the official Edison getting started guide](https://software.intel.com/
         tar -zxf boost_1_58_0.tar.gz
         cd boost_1_58_0/
         ./bootstrap.sh
     
     Now we can build Boost libraries for Edison target.
     
-        source /opt/poky-edison/1.6.1/environment-setup-core2-32-poky-linux
-        sed -r 's/(using\s+gcc)(\s+;)/\1: : i586-poky-linux-g++ : <compileflags>-m32 -march=core2 -mtune=core2 -msse3 -mfpmath=sse -mstackrealign -fno-omit-frame-pointer --sysroot=$SDKTARGETSYSROOT\2/g' -i project-config.jam
+        source /opt/poky-edison/1.6/environment-setup-core2-32-poky-linux
+        sed -r 's#(using\s+gcc)(\s+;)#\1 :  : i586-poky-linux-g++ : <compileflags>-m32 -march=core2 -mtune=core2 -msse3 -mfpmath=sse -mstackrealign -fno-omit-frame-pointer --sysroot='${SDKTARGETSYSROOT}'\2#g' -i project-config.jam
         ./b2 install --prefix=$SDKTARGETSYSROOT/usr/local
 
     Copy `${SDKTARGETSYSROOT}/usr/local/lib/libboost_*` object files to Edison's `/usr/local/lib` directory (e.g. using SSH, SCP, etc.).
 
-2. Compile Avro. Also, Avro depends on some Boost components. So they need to be installed too.
+1. Compile Avro. Also, Avro depends on some Boost components. So they need to be installed too.
 
+        cd $EDISON_ROOT
         mkdir -p avro/target
         tar -xvzf avro-cpp-1.7.5.tar.gz -C ./avro/target
         cd ./avro/target/avro-cpp-1.7.5
-
-        wget https://github.com/kaaproject/kaa/raw/develop/client/client-multi/client-cpp/toolchains/edison.cmake
+        wget https://github.com/kaaproject/kaa/raw/master/client/client-multi/client-cpp/toolchains/edison.cmake
         wget https://raw.githubusercontent.com/kaaproject/kaa/967970ec57fb0a62c23ffe573385bf0d0299d977/client/client-multi/client-cpp/tools/avro-cpp-disable-tests.patch
         patch < avro-cpp-disable-tests.patch
         mkdir build && cd build
-        source /opt/poky-edison/1.6.1/environment-setup-core2-32-poky-linux
-        cmake -DCMAKE_INSTALL_PREFIX=$SDKTARGETSYSROOT -DCMAKE_TOOLCHAIN_FILE=../edison.cmake -DEDISON_SDK_ROOT=/opt/poky-edison/1.6.1 ..
+        source /opt/poky-edison/1.6/environment-setup-core2-32-poky-linux
+        cmake -DCMAKE_INSTALL_PREFIX=$SDKTARGETSYSROOT -DCMAKE_TOOLCHAIN_FILE=../edison.cmake -DEDISON_SDK_ROOT=/opt/poky-edison/1.6 ..
         make && make install
         
     Copy Avro objects files `${SDKTARGETSYSROOT}/usr/lib/libavrocpp*` to `/usr/lib` directory of Edison board.
         
-6. Build Botan.
+1. Build Botan.
 
-        source /opt/poky-edison/1.6.1/environment-setup-core2-32-poky-linux
+        cd $EDISON_ROOT
+        source /opt/poky-edison/1.6/environment-setup-core2-32-poky-linux
         wget http://botan.randombit.net/releases/Botan-1.11.27.tgz
         tar -xvzf Botan-1.11.27.tgz
         cd Botan-1.11.27
-        python configure.py --cpu=x86_32 --cc-bin=${CROSS_COMPILE}g++ --prefix=${SDKTARGETSYSROOT}/usr
+        /usr/bin/python -E configure.py --cpu=x86_32 --cc-bin=${CROSS_COMPILE}g++ --prefix=${SDKTARGETSYSROOT}/usr
```

Comment:
Sysroot should never end up in user's PATH because you never want (or can) to run these tools from a host.

There are many legitimate reasons for having `python` in a different location even on Ubuntu (e.g., `/usr/local/bin/python`), so I'm not buying that.